nomad/structs/structs.go

```diff
@@ -10013,6 +10019,9 @@ func (v *Vault) Copy() *Vault {
 }
 
 func (v *Vault) Canonicalize() {
+	// The Vault cluster name is canonicalized in the jobVaultHook during job
+	// registration because the value may be read from the server config.
```

Comment:
Good call adding the comment so we don't come back and break it later :grinning: